### DISCUSSION TITLE
Enhance FIO to create a nested directory structure...

### DISCUSSION THREAD

**Discussion by elliottecton** [axboe/fio#1675] [Ideas]
I really want to move from `vdbench` to `fio`, but there is one thing that is preventing that transition -- the ability to create a directory structure with `fio`. The storage I am testing has ingest heuristics that will put different sibling directories on different physical volumes. If I put all of the files in a single directory, it is almost guaranteed the files in that directory will all reside on the same volume, limiting the affinities available for IO.

Is it on the roadmap to enable something similar to `vdbench`'s `width` and `depth` parameters?

**Comment by sitsofe** [comment]
Hello @elliottecton:

TLDR; no one "internal" is working on a feature like vdbench's `width` or `depth` but patches are welcome!

As you probably guessed by now it's not on the roadmap but that's because there is no roadmap for for fio :-) Most new fio features are driven by someone showing up with a patch or pull request adding it.
